common/beerocks/bwl/dummy/nl80211_client_dummy.cpp

```diff
@@ -9,13 +9,27 @@
 #include "nl80211_client_dummy.h"
 
 #include <bcl/beerocks_utils.h>
+#include <bcl/network/network_utils.h>
 
 namespace bwl {
 
 nl80211_client_dummy::nl80211_client_dummy() {}
 
 nl80211_client_dummy::~nl80211_client_dummy() {}
 
+bool nl80211_client_dummy::get_interface_info(const std::string &interface_name,
+                                              interface_info &interface_info)
+{
+    interface_info.name  = interface_name;
+    interface_info.index = 1;
+    interface_info.addr  = beerocks::net::network_utils::mac_from_string("01:02:03:04:05:06");
+    interface_info.ssid  = "prplmesh";
```

Comment:
This is a bit problematic. In dummy mode we update the ssid in the `m_radio_info.available_vaps[vap_id].ssid`, so this doesn't reflect that.
Actually I don't understand now why we need a dummy nl80211_client at all - the nl80211_client member is now in the respective BWL base class backend - `base_bwl_wlan_hal_dwapl` and `base_bwl_wlan_hal_nl80211`, but we don't use it in `base_bwl_wlan_hal_dummy` - and we just implement the APIs that we need without it in the dummy implementation.

Seeing how it is now implemented, I think the whole nl80211 client dummy implementation is redundant and potentially confusing.